core/chains/evm/txm/storage/inmemory_store.go

```diff
@@ -0,0 +1,333 @@
+package storage
+
+import (
+	"errors"
+	"fmt"
+	"math/big"
+	"sort"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/smartcontractkit/chainlink-common/pkg/logger"
+	"github.com/smartcontractkit/chainlink/v2/core/chains/evm/txm/types"
+)
+
+const (
+	maxQueuedTransactions = 250
+	pruneSubset           = 3
+)
+
+type InMemoryStore struct {
+	sync.RWMutex
+	lggr      logger.Logger
+	txIDCount uint64
+	address   common.Address
+	chainID   *big.Int
+
+	UnstartedTransactions   []*types.Transaction
+	UnconfirmedTransactions map[uint64]*types.Transaction
+	ConfirmedTransactions   map[uint64]*types.Transaction
+	FatalTransactions       []*types.Transaction
+
+	Transactions map[uint64]*types.Transaction
+}
+
+func NewInMemoryStore(lggr logger.Logger, address common.Address, chainID *big.Int) *InMemoryStore {
+	return &InMemoryStore{
+		lggr:                    logger.Named(lggr, "InMemoryStore"),
+		address:                 address,
+		chainID:                 chainID,
+		UnconfirmedTransactions: make(map[uint64]*types.Transaction),
+		ConfirmedTransactions:   make(map[uint64]*types.Transaction),
+		Transactions:            make(map[uint64]*types.Transaction),
+	}
+}
+
+func (m *InMemoryStore) AbandonPendingTransactions() {
+	m.Lock()
+	defer m.Unlock()
+
+	for _, tx := range m.UnstartedTransactions {
+		tx.State = types.TxFatalError
+	}
+	m.FatalTransactions = m.UnstartedTransactions
```

Comment:
Shouldn't we be appending the `UnstartedTransactions` to the `FatalTransactions` here in case there are fatal transactions stored from another flow?

Reply:
Yeah, but for this stage there was no expectation for persistence, especially for fatal transactions, so I decided to go with a QnD approach. I can update this on the next iteration.